### DISCUSSION TITLE
New Source: Firestore

### DISCUSSION THREAD

**Discussion by Khamaldeeen** [airbytehq/airbyte#35265] [New Connector Request]
## I would like to have an integraiton between firestore database and power bi
* Firestore and local instance (JSON format)
* No 

## Describe the context around this new integration
* Power BI integration. 
* 12 hours
* Not an API source connector

## Describe the alternative you are considering or using
Continue to manually export the data from the firestore to my local instance in a JSON format and load the wrangled files into power bi

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368166086052) by [Unito](https://www.unito.io)

**Comment by marcosmarxm** [comment]
hey @Khamaldeeen thanks for the request! I added this source to our roadmap. Do you want to contribute creating this connector? I can help with the initial steps

**Comment by Khamaldeeen** [comment]
Hi @marcosmarxm  that would be fantastic. I will be glad to share my ideas.

**Comment by cheyura** [comment]
**Webhook-based? (no/partially/yes)**
Partially
https://firebase.google.com/docs/functions/firestore-events


**Available authentication modes (API key/Oauth/other)**
While the easiest way to use Cloud Firestore is to use one of the native client libraries, there are some situations when it is useful to call the REST API directly.

The REST API can be helpful for the following use cases:

Accessing Cloud Firestore from a resource-constrained environment, such as an internet of things (IoT) device, where running a complete client library is not possible.
Automating database administration or retrieving detailed database metadata.


Creating an account
Cloud Firestore offers free quota that allows you to get started at no cost. The free quota amounts are listed below.
https://firebase.google.com/docs/firestore/quotas


Sandbox account?
No


How to populate the account with data?
API,UI


Available streams for sync
Below are brief descriptions of the two most important method groups. For a complete list, see the REST API reference or use the API Explorer.

v1.projects.databases.documents
Perform CRUD operations on documents, similar to those outlined in the add data or get data guides.

v1.projects.databases.collectionGroups.indexes
Perform actions on indexes such as creating new indexes, disabling an existing index, or listing all current indexes. Useful for automating data structure migrations or synchronizing indexes between projects.

Also enables retrieval of document metadata, such as the list of all fields and subcollections for a given document.

**Integration supports incremental sync?**
No

**Other information/blockers**
No

**Comment by julienfouilhe** [comment]
@igrankova I'm sorry to comment on a closed issue, but it's been closed as completed but I can't find the firestore source in Airbyte's list, is that normal?
I'd be happy to contribute a Firestore source but I wanted to ensure it's not already done. Thanks!

**Comment by igrankova** [comment]
hi, @julienfouilhe, as I see, we have a destination Firestore, but no source. 
@YowanR we have a new contribution offered. can you help with the steps what to do, because I'm not in this process.

**Comment by lightstrike** [comment]
@julienfouilhe I'm also looking to create a Firestore source, let me know if you want to collaborate on this!

**Comment by folkevil** [comment]
Hi guys,  have any update for Firebase source ? I see the document in website but can't find repo for this source.

**Comment by andremosko** [comment]
Hi! Any updates for Firestore as a source? Thanks =)

[a comment by an automated account is not shown]
